_posts/2020-12-11-an-old-new-adventure.md

```diff
@@ -0,0 +1,61 @@
+
+## On open source sustainability
+
+Much has been written on open source sustainability, notably [Nadia Eghbal]'s *Working in Public.* I won't speak to open source more broadly (except to note how impressed I am with Blender and Krita), but for the specific task of building an ecosystem for a library, I think there is one model that actually works: being hired by a company that depends on that ecosystem.
+
+To some extent, that's an indictment of our capitalist system. In an ideal universe, there would be strong institutions dedicated to the public interest where open source developers could develop, researchers could research, and spend an absolute minimum of time and energy hustling for support. For software, in any case, universities are [not that,][stein on leaving ucsd] otherwise I'd be quite tempted. In the actual world, working for a company like Google is about as close as you can come.
+
+I remain skeptical of patronage-style platforms such as Patreon or Github Sponsors. I think it's possible to make them work, but only for a small number of fortunate people, and even then, the incentives for creating maximum value aren't that well aligned with the incentive structure of hustling on social media.
+
+So me (re-)joining Google full time is basically a statement of confidence in the model of being employed to work on open source.
+
+## On Rust
+
+I continue to love Rust, and believe it offers a stronger foundation for building software. I feel like I started my Rust journey in the early '90s, when I was working on retrofitting [static memory management] to ML, using explicit lifetime regions.
+
+Rust adoption is trending up, including at Google. The language is in good shape, but the library ecosystem is still fairly immature, missing a number of critical pieces. Building up that ecosystem is an incredibly rewarding project.
+
+I am particularly excited about Rust for font technology and infrastructure. Today, Python rules on the font design and production side, partly to the connection of typeface designer [Just van Rossum] being Guido's brother. The flexibility and expressiveness of Python makes it a good fit, but we've also gotten to a place where the *production* of fonts is done in Python, and the *consumption* is in C++.
+
+Rust lets us build reliable, performant code that can also be deployed in production, and can be the basis of fluidly interactive UI tools. I'm not the only one who sees this potential; YesLogic is building their next-generation [font shaper][Allsorts] in Rust, for many of the same reasons.
```

Comment:
Similar remark to earlier, I think this isn’t good link text. I think “YesLogic is building their next-generation font shaper [Allsorts] in Rust” would be better.